### PR TITLE
Consistent usage of es6 features in modules - Closes #1680

### LINKED ISSUE #1680: Consistent usage of es6 features in modules implementation
Parent #1709
### Expected behavior
Use of es6 features in modules repo to make sure consistency and making use of the latest features.
### Actual behavior
Most of the modules implementations are not inline with the es6 features.
- [x] accounts.js
- [x] blocks.js
- [x] cache.js
- [x] dapps.js
- [x] delegates.js
- [x] loader.js
- [x] multisignatures.js
- [x] node.js
- [x] peers.js
- [x] rounds.js
- [x] signatures.js
- [x] system.js
- [x] transactions.js
- [x] transport.js
- [x] voters.js

blocks folder
- [x] api.js 
- [x] chain.js
- [x] process.js
- [x] utils.js
- [x] verify.js (waiting #1657 )
### Which version(s) does this affect? (Environment, OS, etc...)
1.0.0

### PR DESCRIPTION
**Pull request by LucasIsasmendi**
### What was the problem?
Folder `modules` was in ES5
### How did I fix it?
Code migration to ES6: 
- variables (const, let) 
- constructor class
### Review checklist

* The PR solves #1680 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: modules/blocks/process.js [hunk 1]
```diff
@@ -177,7 +179,7 @@ __private.receiveForkOne = function(block, lastBlock, cb) {
  * @param {function} cb - Callback function
  */
 __private.receiveForkFive = function(block, lastBlock, cb) {
-	var tmp_block = _.clone(block);
+	let tmp_block = _.clone(block);
```
Comment (ManuGowda) [inline modules/blocks/process.js]: Change it to camelCase `tmpBlock` to follow consistency.

File: modules/blocks/verify.js [hunk 2]
```diff
@@ -300,14 +301,14 @@ __private.verifyPayload = function(block, result) {
...
-	var appliedTransactions = {};
+	let totalAmount = 0;
+	let totalFee = 0;
+	const payloadHash = crypto.createHash('sha256');
+	const appliedTransactions = {};
 
-	for (var i in block.transactions) {
-		var transaction = block.transactions[i];
-		var bytes;
+	for (const i in block.transactions) {
```
Comment (ManuGowda) [inline modules/blocks/verify.js]: `For-of` method is a more reliable way of looping through an Array in sequence.
FYI: https://bitsofco.de/for-in-vs-for-of/

File: modules/delegates.js [hunk 3]
```diff
@@ -438,10 +439,10 @@ __private.checkDelegates = function(publicKey, votes, state, cb, tx) {
 						return setImmediate(cb, err);
 					}
 
-					var total_votes = existing_votes + additions - removals;
+					const total_votes = existing_votes + additions - removals;
```
Comment (ManuGowda) [inline modules/delegates.js]: camelCase.

File: modules/delegates.js [hunk 4]
```diff
@@ -365,16 +366,16 @@ __private.checkDelegates = function(publicKey, votes, state, cb, tx) {
 				return setImmediate(cb, 'Account not found');
 			}
 
-			var delegates =
+			const delegates =
 				state === 'confirmed' ? account.delegates : account.u_delegates;
-			var existing_votes = Array.isArray(delegates) ? delegates.length : 0;
-			var additions = 0;
-			var removals = 0;
+			const existing_votes = Array.isArray(delegates) ? delegates.length : 0;
```
Comment (ManuGowda) [inline modules/delegates.js]: camelCase.

File: modules/peers.js [hunk 5]
```diff
@@ -119,9 +119,9 @@ __private.getByFilter = function(filter, cb) {
 	 * @todo Add @returns tag
 	 * @todo Add description of the function
 	 */
-	var sortPeers = function(field, asc) {
+	const sortPeers = function(field, asc) {
 		return function(a, b) {
-			var sort_res =
+			const sort_res =
```
Comment (ManuGowda) [inline modules/peers.js]: camelCase

**Comment by ManuGowda** [comment]
@MaciejBaj @LucasIsasmendi the es6 changes made to `blocks`, `transactions`, `transport` causes the unit, functional and integrations tests to fail, changing `var` to `let` and `const` should not cause any impact, since the way we have written our test to `mock` and reassign the values inside module casing the tests to fail. In order to introduce es6 features into `blocks`, `transactions`, `transport` we need to revisit all the unit, functional and integrations tests, for now, I have reverted the es6 changes made to `blocks`, `transactions`, `transport` modules.

File: test/unit/modules/peers.js [hunk 6]
```diff
@@ -927,7 +926,7 @@ describe('peers', () => {
 		});
 
 		afterEach(() => {
-			return revertPrivateStubs();
+			return revertPrivateStubs;
```
Comment (MaciejBaj) [inline test/unit/modules/peers.js]: What does this part do now? Is it even needed?

Comment (ManuGowda) [reply test/unit/modules/peers.js]: I think the variable it self is unused, for some reason @LucasIsasmendi changed it, but i am removing this variable and afterEach hook as the variable is not being used anywhere and all the tests are working.

**Pull request merged**